eclass/toolchain.eclass

```diff
@@ -1000,8 +1000,17 @@ toolchain_setup_ada() {
 # Determine the most suitable GDC (D compiler) for bootstrapping
 # and setup the environment for building.
 toolchain_setup_d() {
-	local latest_gcc=$(best_version -b "sys-devel/gcc")
-	latest_gcc="${latest_gcc#sys-devel/gcc-}"
+	local gcc_pkg gcc_bin_base
```

Comment:
If possible, it would be nice to keep this in sync with the Ada logic but i won't insist on it here.

Reply:
I'll see what I can do but what's the support for cross-compiling with ada? I'm guessing you can't have a native non-amd64 compiler as the `ada` USE flag is only unmasked on amd64/x86 but from your comment in `toolchain_setup_ada` I'm inferring that you can install `cross-${CTARGET}/gcc[ada]` on amd64 and have that compile ada programs for non-amd64.

For building `cross-${CTARGET}/gcc[ada|d]` would it be fine to require that `sys-devel/gcc[ada|d]` be installed or do you want me to also support any of the bootstrap paths that are available for `sys-devel/gcc`? So, _if_ the code would get complex, is it fine to drop the checks for `ada-bootstrap` et all? I think we should also consider the future possibility of our bootstrap tarballs.

Reply:
I've not tried it, I'd like it to work, and I suspect it _might_ work at least with some minor fiddling as you did here.

But yes, the problem is the lack of bootstrap tarballs right now for other arches, which means while Ada should work on non-amd64/x86, getting an Ada there in the first place requires cross.

I think it's OK to have the requirement you described.